src/CarbonAware.WebApi/test/unitTests/TestsBase.cs

```diff
@@ -20,60 +18,63 @@ protected TestsBase()
         this.MockCarbonAwareLogger = new Mock<ILogger<CarbonAwareController>>();
     }
 
-    protected static Mock<IEmissionsAggregator> CreateEmissionsAggregator(List<EmissionsData> data)
+    protected static Mock<IEmissionsHandler> CreateEmissionsHandler(List<EmissionsData> data)
     {
-        var aggregator = new Mock<IEmissionsAggregator>();
-        aggregator.Setup(x => x.GetEmissionsDataAsync(It.IsAny<CarbonAwareParameters>()))
-            .Callback((CarbonAwareParameters parameters) =>
+        var handler = new Mock<IEmissionsHandler>();
+
+        handler.Setup(x => x.GetEmissionsDataAsync(It.IsAny<string[]>(), null, null))
+            .Callback((string[] location, DateTimeOffset? start, DateTimeOffset? end) =>
             {
-                parameters.SetRequiredProperties(PropertyName.MultipleLocations);
-                parameters.Validate();
+                Assert.NotNull(location);
+                Assert.NotZero(location.Length);
             })
             .ReturnsAsync(data);
-        return aggregator;
+        return handler;
     }
 
-    protected static Mock<IEmissionsAggregator> CreateAggregatorWithBestEmissionsData(List<EmissionsData> data)
+    protected static Mock<IEmissionsHandler> CreateHandlerWithBestEmissionsData(List<EmissionsData> data)
     {
-        var aggregator = new Mock<IEmissionsAggregator>();
-        aggregator.Setup(x => x.GetBestEmissionsDataAsync(It.IsAny<CarbonAwareParameters>()))
-            .Callback((CarbonAwareParameters parameters) =>
+        var handler = new Mock<IEmissionsHandler>();
+        handler.Setup(x => x.GetBestEmissionsDataAsync(It.IsAny<string[]>(), null, null))
+            .Callback((string[] location, DateTimeOffset? start, DateTimeOffset? end) =>
             {
-                parameters.SetRequiredProperties(PropertyName.MultipleLocations);
-                parameters.Validate();
+                Assert.NotNull(location);
```

Comment:
Is this a full accounting of everything the Validation step was doing? I recall that we had some significant issues that were resolved by adding that step and want to be double sure we aren't re-introducing resolved issues.